app/assets/javascripts/spree/frontend/solidus_paypal_commerce_platform/button_actions.js

```diff
@@ -11,6 +11,56 @@ SolidusPaypalCommercePlatform.postOrder = function(payment_method_id) {
   })
 }
 
+SolidusPaypalCommercePlatform.createAndSendOrder = function(payment_method_id) {
+  return SolidusPaypalCommercePlatform.createOrder().then(function(){
+    return SolidusPaypalCommercePlatform.sendOrder(payment_method_id)
+  })
+}
+
+SolidusPaypalCommercePlatform.createOrder = function() {
+  var data = {
+    order: {
+      line_items_attributes: [{
+        variant_id: SolidusPaypalCommercePlatform.getVariantId(),
+        quantity: SolidusPaypalCommercePlatform.quantity_field.value
+      }]
+    }
+  }
+
+  if(typeof SolidusPaypalCommercePlatform.user_id !== 'undefined'){
+    data["user_id"] = SolidusPaypalCommercePlatform.user_id
+  }
```

Comment:
Can we get rid of the explicit `user_id` and just rely on the authentication data stored in the cookie?

Reply:
Actually, since we're using a Spree.ajax request, I can just include the auth helper and use `try_spree_current_user` and do away with all the user stuff in the javascript. Thanks!